docs/backend/api-definitions-guidelines.md

```diff
@@ -0,0 +1,58 @@
+
+The `InstructLab APIs` are essential for hosting `InstructLab` as a service in a repeatable way. The `Platform APIs` are critical for component reuse and extensibility (e.g., new SDG algorithms for new taxonomy data types), but they are not strictly required for hosting `InstructLab` as a service.
+
+## How will service APIs be defined?
+
+Service APIs will be defined using [OpenAPI](https://www.openapis.org/) format in [YAML](https://yaml.org/).
+
+## Where will service API definitions live?
+
+Service API definitions will live a new repository github.com/instructlab/service-api-definitions. This repo will have two primary responsibilities:
+
+1. House the static service API definitions
+2. Build and publish any language-specific generated packages for consumption by service implementation projects (see below)
+
+## How will service implementations reference shared APIs?
+
+When a project chooses to implement one or more service APIs, there are three acceptable methods for doing so, listed in order of preference:
+
+1. Consume a supported language-specific package. The `service-api-definitions` repo will build consumable packages with generated code for supported languages. This is the preferred method of consumption as it avoids repository references and code duplication.
+2. For languages without a supported package, the `service-api-definitions` repo may be held as a [git submodule](https://www.git-scm.com/book/en/v2/Git-Tools-Submodules).
+3. It is also acceptable for an implementation to copy the relevant API definitions to the local project repository. Any changes made in the central repository will need to be sync'ed by the project owners, and any new APIs added in the project will not be considered usable until they have been integrated into the central API definitions.
+
+## Style Guidelines
+
+* Use `kebab-case` for path elements
+  * All characters must be in the [ascii](https://www.ascii-code.com/) character set to avoid percent encoding in URIs
+  * All letters must be lowercase
+  * Words are separated by the `-` (dash) character
+* Use `snake_case` for properties
+  * All characters must be in the [utf-8](https://www.w3schools.com/charsets/ref_html_utf8.asp) character set for simple `json` encoding
+  * Words are separated by the `_` (underscore) character
+* Use `UpperCamelCase` for internal reusable schema names
+  * These are internal names, so the character set is not limited
+  * Words are capitalized and concatenated with no separator
+
+## API Layout
+
+* There will be two main portions of the APIs:
+  * `instructlab.yaml`: This defines the user-facing `InstructLab` REST API
+  * `platform.yaml`: This defines the platform-level APIs used by the `InstructLab` workflow.
+* Each platform `Capability` should own its own fully-functional sub-API file that can be used by individual capability service implementations
```

Comment:
what is "platform capability"? is it defined elsewhere? maybe an example could clarify it.

Reply:
I somewhat defined it above under [What parts of InstructLab need service APIs](https://github.com/instructlab/dev-docs/pull/71/files#diff-5d6426ec3d00a810ecc49e2fc141ac72545910a7caaaaccbd0ab42315324c96eR9). It's admittedly a bit of a "definition by example." Does this feel sufficient?

Reply:
Is `capability` here just a `component` of sorts? The `capability` term threw me off a bit (I think of capabilities as properties of things; not as things themselves.)

Reply:
Yeah, I'm using this word a bit loosely, though we've been working on a more rigorous definition outside of the InstructLab scope.

My other hat outside of IL has been focusing on trying to put shape around "the AI platform." To this end, I've wanted a way to separate _what_ the platform can do from _how_ it does it (interface vs implementation). I've been using the word `capability` to mean a logical grouping of interface definitions that holistically accomplish some user-focused AI need (e.g. "fine tune a model" would include "submit fine tuning," "poll tuning job," "cancel tuning job," etc...). Each `capability` can be implemented using any number of projects, and may have multiple implementations depending on use-case (e.g. multiple SDGs for different form-factors).

In this sense the `capability` is a property of the `platform`. I didn't really want to lead with trying to define abstract platform terminology in this PR, but if there's appetite, I'd be happy to try to put something together to formalize the verbiage.

Reply:
Thanks Gabe for quick answer. Terms are important. But no need to expand this PR. I was just confused. I think I understand where you come from, even if I may not be too thrilled about particular word choices.